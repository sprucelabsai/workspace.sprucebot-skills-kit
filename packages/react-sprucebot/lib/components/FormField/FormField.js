'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});
exports.FieldLabel = exports.FieldError = exports.FieldWrapper = undefined;
exports.default = Field;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _reactTextareaAutosize = require('react-textarea-autosize');

var _reactTextareaAutosize2 = _interopRequireDefault(_reactTextareaAutosize);

var _Typography = require('../Typography/Typography');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var FieldWrapper = exports.FieldWrapper = _styledComponents2.default.div.attrs({
	className: (0, _classnames2.default)('InputField', 'input__wrapper')
}).withConfig({
	displayName: 'FormField__FieldWrapper',
	componentId: 'q2a077-0'
})(['']);

var FieldError = exports.FieldError = _styledComponents2.default.span.attrs({
	className: (0, _classnames2.default)('FieldError', 'input__error', 'error-is-visible')
}).withConfig({
	displayName: 'FormField__FieldError',
	componentId: 'q2a077-1'
})(['']);

var FieldLabel = exports.FieldLabel = _styledComponents2.default.span.attrs({
	className: (0, _classnames2.default)('FieldLabel', 'input__mini__label', 'js-show-label')
}).withConfig({
	displayName: 'FormField__FieldLabel',
	componentId: 'q2a077-2'
})(['']);

function Field(_ref) {
	var _ref$input = _ref.input,
	    value = _ref$input.value,
	    onChange = _ref$input.onChange,
	    name = _ref$input.name,
	    _ref$meta = _ref.meta,
	    touched = _ref$meta.touched,
	    error = _ref$meta.error,
	    type = _ref.type,
	    finePrint = _ref.finePrint,
	    label = _ref.label,
	    _ref$placeholder = _ref.placeholder,
	    placeholder = _ref$placeholder === undefined ? label : _ref$placeholder,
	    children = _ref.children;

	return _react2.default.createElement(
		FieldWrapper,
		null,
		label && value && _react2.default.createElement(
			FieldLabel,
			null,
			label
		),
		_react2.default.cloneElement(children, {
			className: (0, _classnames2.default)({ with_label: !!label }),
			name: name,
			value: value,
			onChange: onChange,
			type: type,
			placeholder: placeholder
		}),
		touched && error && _react2.default.createElement(
			FieldError,
			null,
			error
		),
		finePrint && _react2.default.createElement(
			_Typography.Paragraph,
			{ fine: true },
			finePrint
		)
	);
}

Field.propTypes = {
	input: _propTypes2.default.shape({
		value: _propTypes2.default.any.isRequired,
		onChange: _propTypes2.default.func.isRequired
	}).isRequired,
	meta: _propTypes2.default.shape({
		touched: _propTypes2.default.bool,
		error: _propTypes2.default.string
	}).isRequired,
	children: _propTypes2.default.element.isRequired
};