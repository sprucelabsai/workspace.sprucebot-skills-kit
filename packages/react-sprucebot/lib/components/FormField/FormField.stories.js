'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _react3 = require('@storybook/react');

var _addonInfo = require('@storybook/addon-info');

var _react4 = require('@storybook/addon-knobs/react');

var _storybookReadme = require('storybook-readme');

var _FormField = require('./FormField');

var _FormField2 = _interopRequireDefault(_FormField);

var _FormField3 = require('./FormField.md');

var _FormField4 = _interopRequireDefault(_FormField3);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var stories = (0, _react3.storiesOf)('FormField', module);
stories.addDecorator(_react4.withKnobs);

stories.add('Interactive', (0, _storybookReadme.withReadme)(_FormField4.default, (0, _addonInfo.withInfo)()(function () {
	return _react2.default.createElement(
		_FormField2.default,
		{
			input: { value: '', onChange: function onChange() {
					return null;
				} },
			meta: { touched: false, error: null },
			type: 'text',
			label: 'Email Address',
			placeholder: 'hello@sprucebot.com'
		},
		_react2.default.createElement('input', {
			type: 'text',
			label: 'Email Address',
			placeholder: 'hello@sprucebot.com'
		})
	);
})));