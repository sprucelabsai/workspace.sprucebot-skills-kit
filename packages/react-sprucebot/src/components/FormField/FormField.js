import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import classnames from 'classnames'
import TextArea from 'react-textarea-autosize'

import { Paragraph as P } from '../Typography/Typography'

export const FieldWrapper = styled.div.attrs({
	className: classnames('InputField', 'input__wrapper')
})``

export const FieldError = styled.span.attrs({
	className: classnames('FieldError', 'input__error', 'error-is-visible')
})``

export const FieldLabel = styled.span.attrs({
	className: classnames('FieldLabel', 'input__mini__label', 'js-show-label')
})``

export default function Field({
	input: { value, onChange, name },
	meta: { touched, error },
	type,
	finePrint,
	label,
	placeholder = label,
	children
}) {
	return (
		<FieldWrapper>
			{label && value && <FieldLabel>{label}</FieldLabel>}
			{React.cloneElement(children, {
				className: classnames({ with_label: !!label }),
				name,
				value,
				onChange,
				type,
				placeholder
			})}
			{touched && error && <FieldError>{error}</FieldError>}
			{finePrint && <P fine>{finePrint}</P>}
		</FieldWrapper>
	)
}

Field.propTypes = {
	input: PropTypes.shape({
		value: PropTypes.any.isRequired,
		onChange: PropTypes.func.isRequired
	}).isRequired,
	meta: PropTypes.shape({
		touched: PropTypes.bool,
		error: PropTypes.string
	}).isRequired,
	children: PropTypes.element.isRequired
}
