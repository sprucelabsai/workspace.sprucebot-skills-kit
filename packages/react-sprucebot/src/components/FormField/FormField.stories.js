import React from 'react'
import { storiesOf } from '@storybook/react'
import { withInfo } from '@storybook/addon-info'
import { withKnobs } from '@storybook/addon-knobs/react'
import { withReadme } from 'storybook-readme'
import FormField from './FormField'
import readme from './FormField.md'

const stories = storiesOf('FormField', module)
stories.addDecorator(withKnobs)

stories.add(
	'Interactive',
	withReadme(
		readme,
		withInfo()(() => (
			<FormField
				input={{ value: '', onChange: () => null }}
				meta={{ touched: false, error: null }}
				type="text"
				label="Email Address"
				placeholder="hello@sprucebot.com"
			>
				<input
					type="text"
					label="Email Address"
					placeholder="hello@sprucebot.com"
				/>
			</FormField>
		))
	)
)
