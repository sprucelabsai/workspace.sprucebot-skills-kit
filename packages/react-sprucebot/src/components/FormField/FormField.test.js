import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import FormField from './FormField'
const onChange = jest.fn()

test('it renders set value and triggers onChange', () => {
	const wrapper = shallow(
		<FormField
			type="text"
			input={{ value: 'test1', onChange }}
			meta={{ touched: false }}
		>
			<input />
		</FormField>
	)
	expect(wrapper).toMatchSnapshot()
	wrapper
		.find('input[type="text"]')
		.simulate('change', { target: { value: 'test' } })
	expect(onChange).toHaveBeenCalledWith({ target: { value: 'test' } })
})

test('it renders with a label, finePrint, placeholder, and error', () => {
	const wrapper = shallow(
		<FormField
			label="Test Label"
			placeholder="Placeholdn'"
			type="text"
			input={{ value: 'test2', onChange }}
			meta={{ touched: true, error: 'TEST ERROR' }}
			finePrint="Fine Print!"
		>
			<input />
		</FormField>
	)
	expect(wrapper).toMatchSnapshot()
})
