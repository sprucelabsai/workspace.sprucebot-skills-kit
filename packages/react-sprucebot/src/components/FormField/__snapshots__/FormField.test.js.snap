// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders set value and triggers onChange 1`] = `
<div
  className="InputField input__wrapper"
>
  <input
    className=""
    onChange={[MockFunction]}
    type="text"
    value="test1"
  />
</div>
`;

exports[`it renders with a label, finePrint, placeholder, and error 1`] = `
<div
  className="InputField input__wrapper"
>
  <span
    className="FieldLabel input__mini__label js-show-label"
  >
    Test Label
  </span>
  <input
    className="with_label"
    onChange={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "target": Object {
                "value": "test",
              },
            },
          ],
        ],
      }
    }
    placeholder="Placeholdn'"
    type="text"
    value="test2"
  />
  <span
    className="FieldError input__error error-is-visible"
  >
    TEST ERROR
  </span>
  <Paragraph
    fine={true}
  >
    Fine Print!
  </Paragraph>
</div>
`;
